htdocs/product/dynamic_price/class/price_parser.class.php

```diff
@@ -133,7 +133,16 @@ public function translatedError()
     public function parseExpression($product, $expression, $values)
     {
         global $user;
-
+        global $hookmanager;
+        $hookmanager->initHooks(array('productcard','globalcard'));
```

Comment:
I don't think we should initialize the context of the hook here. parseExpression can be called at any place in a lot of other context.
What if you remove this line completely ?

Reply:
Any suggestion?

Reply:
Just remove the line $hookmanager->initHooks(array('productcard','globalcard'));

Reply:
there isn´t any hook loaded any times. I can add a condition
                if (!in_array('productcard', $hookmanager->hooks)) {
                    $hookmanager->initHooks(array('productcard'));
                }

Reply:
Do you want to have the hook called only when the parseExpression is called from the web page of product record or do you need it any time the parseExpression is called. I think it is the second choice, isn't it ?
You should have the context "main" that is always enabled.

Reply:
true, its work, thank you very much